cmd/goal/box.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var boxName string
+
+func init() {
+	appCmd.AddCommand(appBoxCmd)
+
+	appBoxCmd.AddCommand(appBoxInfoCmd)
+	appBoxCmd.AddCommand(appBoxListCmd)
+	appBoxCmd.PersistentFlags().Uint64Var(&appIdx, "app-id", 0, "Application ID")
+	appBoxCmd.MarkFlagRequired("app-id")
+
+	appBoxInfoCmd.Flags().StringVarP(&boxName, "name", "n", "", "Application box name. Use the same form as app-arg to name the box.")
+	appBoxInfoCmd.Flags().SetInterspersed(false)
+	appBoxInfoCmd.MarkFlagRequired("name")
+}
+
+var appBoxCmd = &cobra.Command{
+	Use:   "box",
+	Short: "Read application box data",
+	Args:  cobra.ArbitraryArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		// If no arguments passed, we should fallback to help
+		cmd.HelpFunc()(cmd, args)
+	},
+}
+
+var appBoxInfoCmd = &cobra.Command{
+	Use:   "info",
+	Short: "Retrieve information about an application box",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		_, client := getDataDirAndClient()
+
+		// Ensure box name is specified
+		if boxName == "" {
+			reportErrorf(errorMissingBoxName)
+		}
+
+		// Get box info
+		box, err := client.GetApplicationBoxByName(appIdx, boxName)
+		if err != nil {
+			reportErrorf(errorRequestFail, err)
+		}
+
+		// Print box info
+		reportInfof("Name:  %s", box.Name)
+		reportInfof("Value: %s", box.Value)
+	},
+}
+
+var appBoxListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List all application boxes belonging to an application",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		_, client := getDataDirAndClient()
+
+		// Get app boxes
+		boxes, err := client.ApplicationBoxes(appIdx)
+		if err != nil {
+			reportErrorf(errorRequestFail, err)
+		}
+
+		// Print if no boxes found
+		if len(boxes) == 0 {
+			reportInfof("No application boxes found")
```

Comment:
nit: I wonder if we can create a box called "No application boxes found", and would that confuse with this line report?